.github/workflows/build_release.yml

```diff
@@ -58,32 +58,23 @@ jobs:
       run: find bin/ -type f -perm -u=x -exec strip {} +
       if: matrix.os != 'windows-latest'
 
-    - name: archive
-      id: archive
-      run: |
-        OSNAME=$(echo ${{ matrix.os }} | sed 's/-latest//')
-        VERSION=${{ github.event.release.tag_name }}
-        PKGNAME="wabt-$VERSION-$OSNAME"
-        TARBALL=$PKGNAME.tar.gz
-        SHASUM=$PKGNAME.tar.gz.sha256
-        mv install wabt-$VERSION
-        tar -czf $TARBALL wabt-$VERSION
-        scripts/sha256sum.py $TARBALL > $SHASUM
-        echo "::set-output name=tarball::$TARBALL"
-        echo "::set-output name=shasum::$SHASUM"
+    - uses: ./.github/actions/upload-release-archive
+      with:
+        os: ${{ runner.os }}
 
-    - name: upload tarball
-      uses: svenstaro/upload-release-action@v2
+  build-wasi:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
       with:
-        repo_token: ${{ secrets.GITHUB_TOKEN }}
-        file: ./${{ steps.archive.outputs.tarball }}
-        asset_name: ${{ steps.archive.outputs.tarball }}
-        tag: ${{ github.ref }}
+        submodules: true
+
+    - name: built-tools
+      run: |
+        docker run -di --name wasi-sdk -v $(pwd):/src --workdir /src ghcr.io/webassembly/wasi-sdk:wasi-sdk-20 bash
```

Comment:
This first line looks like ti run an interactive bash shell.. and the deletes (`-d`) the container afterwards..  what is the purpose of that?

Reply:
`-d` detaches rather than deletes

https://docs.docker.com/engine/reference/commandline/run/

This allows it to stay running for the other `exec` commands. TBH, I just copied the same pattern as the emscripten build but it seems reasonable - I did try just setting the GHA job itself to use wasi-sdk as its container but the lack of a `git` binary caused source checkout to fail

Reply:
sgtm